include/c.h

```diff
@@ -595,4 +595,16 @@ static inline void *reallocarray(void *ptr, size_t nmemb, size_t size)
 }
 #endif
 
+static inline void _reset_errno_(int *saved_errno) {
+        if (*saved_errno < 0)
+                return;
+
+        errno = *saved_errno;
+}
+
+#define _unused_ __attribute__((__unused__))
+#define _cleanup_(x) __attribute__((__cleanup__(x)))
+
```

Comment:
We usually use directly `__attribute__` in the code if there is no portability issue for the attribute. And please use the ul_ prefix for the function and the variable.

I think one macro should be good enough:
```C
#define UL_PROTECT_ERRNO __attribute__((__cleanup__(ul_reset_errno))) \
                         __attribute__((__unused__)) int __ul_saved_errno = errno
```

Reply:
I have updated the PR with your suggestions